### PR TITLE
WIP: Store VTT data in lib

### PR DESCRIPTION
**Pull request by madig**
Tackle https://github.com/daltonmaag/vttLib/issues/19.

Make Python 3.6+ only in the process because ugh.

### CODE REVIEW

**Comment by moyogo** [comment]
This may be nitpicking, the Ubuntu Test Data font shouldn’t have "Ubuntu" in its name. The Ubuntu license prevents derivatives from having "Ubuntu" in their name. The license should be included as well.

**Comment by madig** [comment]
Oops, I forgot. I guess my subsetting qualifies as "Substantially Changed" so I'll change the name.

**Comment by madig** [comment]
This is indeed a slower. As a side-effect, we do not optionally need ufoNormalizer. As the strings in the TSI tables contain `\r` for newlines, I gotta test what editors make of them...

**Comment by madig** [comment]
Round-tripping Ubuntu-B with VTT data in lib keys with glyphsLib now gives me:
```
> glyphs2ufo.exe .\Ubuntu.glyphs
Traceback (most recent call last):
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\nikolaus.waxweiler\AppData\Local\Programs\Python\Python37\Scripts\glyphs2ufo.exe\__main__.py", line 9, in <module>
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\cli.py", line 189, in _glyphs2ufo_entry_point
    return main(args)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\cli.py", line 153, in main
    return options.func(options)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\cli.py", line 181, in glyphs2ufo
    generate_GDEF=options.generate_GDEF,
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\__init__.py", line 109, in build_masters
    font = GSFont(filename)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\classes.py", line 3194, in __init__
    p.parse_into_object(self, fp.read())
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 63, in parse_into_object
    i = self._parse_dict_into_object(res, text, 1)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 164, in _parse_dict_into_object
    result = self._parse(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 101, in _parse
    return self._parse_list(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 191, in _parse_list
    list_item, i = self._parse(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 95, in _parse
    return self._parse_dict(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 148, in _parse_dict
    i = self._parse_dict_into_object(res, text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 164, in _parse_dict_into_object
    result = self._parse(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 95, in _parse
    return self._parse_dict(text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 148, in _parse_dict
    i = self._parse_dict_into_object(res, text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 174, in _parse_dict_into_object
    self._fail("Missing delimiter in dictionary before content", text, i)
  File "c:\users\nikolaus.waxweiler\appdata\local\programs\python\python37\lib\site-packages\glyphsLib\parser.py", line 233, in _fail
    raise ValueError("{}:\n{}".format(message, text[i : i + 79]))
ValueError: Missing delimiter in dictionary before content:
H H H"

/*
Additional Character group declarations

To declare additional chara
```
Looks like custom parameters aren't properly escaped. Continuing tomorrow...

**Comment by anthrotype** [comment]
I would expect the CR character `\r` to be escaped with octal escape by glyphsLib writer, if not it's yet another bug.

**Comment by madig** [comment]
Turns out that glyphsLib does the right thing, however, opening and saving the file in Glyphs.app unescapes the `"` inside the custom parameter content, destroying the file...

**Comment by anthrotype** [comment]
wow

File: src/vttLib/transfer.py [hunk 1]
```diff
@@ -25,7 +25,9 @@ def dump_to_lib_keys(font, ufo):
     for extra_program in vttLib.TSI1_EXTRA_PROGRAMS:
         if extra_program in font_tsi1.extraPrograms:
             lib_key = f"com.daltonmaag.vttLib.tsi1.{extra_program}"
-            ufo.lib[lib_key] = font_tsi1.extraPrograms[extra_program]
+            ufo.lib[lib_key] = font_tsi1.extraPrograms[extra_program].replace(
+                "\r", "\n"
```
Comment (anthrotype) [inline src/vttLib/transfer.py:29]: using splitlines() and then "\n".join() would be safer I think

**Comment by anthrotype** [comment]
but then you need to restore the carriage returns, I think VTT requires those

**Comment by madig** [comment]
I think VTT can handle both or there's some kind of conversion going on behind my back, it works when I store `\n` in the lib key and write that to a TTF.

I asked @schriftgestalt about this, let's see.

**Comment by schriftgestalt** [comment]
Can you send me a .glyphs file that shows the unescaping problem?

**Comment by madig** [comment]
I think I sent you one back then. IIRC, you mentioned somewhere that the problem was fixed in some newer version.

**Comment by madig** [comment]
I am going to store the data in external TTX blobs instead.

**Pull request closed without merging**